### ISSUE TITLE
Cleanup Borsh Serializers

### ISSUE THREAD

**Issue by nearmax** [near/nearcore#2167]
- [x] Allow deriving Borsh schema using `#[derive(BorshSchema)]`. https://github.com/nearprotocol/borsh/pull/56
- [x] Implement Rust contract metadata and use Borsh schema for types;
- [ ] Reconcile and update Borsh for TS/JS (Applayer workspace): https://github.com/nearprotocol/nearlib/issues/252
- [ ] Upgrade rpctypegen to new Borsh Schema format (blocked by the above issue).

**Comment by lazaridiscom** [comment]
Never liked cabbage, until I ate borsh.

Question is if this naming-indirection makes sense. I mean, reading technical docs and getting hungry, and guessing what this library does?

Possibly you should consider giving "borsh" a more expressive name, something that implies what this library does.

near-stream?

**Comment by nearmax** [comment]
> Never liked cabbage, until I ate borsh.
> 
> Question is if this naming-indirection makes sense. I mean, reading technical docs and getting hungry, and guessing what this library does?
> 
> Possibly you should consider giving "borsh" a more expressive name, something that implies what this library does.
> 
> near-stream?

Borsh name is too much ingrained into our codebase to be easily switched. Also, similar to CBOR, BORSH is a meaningful abbreviation see http://borsh.io/

**Comment by nearmax** [comment]
Assigning to @kcole16 since the rest of the work is mostly on the Applayer side.

**Comment by ilblackdragon** [comment]
@nearmax I'm a bit lost given some work was already done - Is the other steps required for MainNet or not?

**Comment by nearmax** [comment]
This is not a Mainnet blocker. Since the remaining steps will not change the node API, we will be able to update via nearlib. Moving to the IceBox and removing from the Mainnet release.

**Comment by bowenwang1996** [comment]
I believe the majority of the work is done. The rest is tracked in #2242
